### ISSUE TITLE
Limit to number of conditional statements on one line?

### ISSUE THREAD

**Issue by kwince** [buildbotics/bbctrl-firmware#232]
As per Camotics page attached as an image, I test a variable that I pass between a probing program and if all three axis are zeroed:

```
o1000 if [  [ #5000 eq 0 ] OR [ #<_x_homed>  EQ 0] OR [ #<_y_homed> EQ 0]  OR [ #<_z_homed> EQ 0]  ]
   (msg, MACHINE NOT HOMED using 'zero.nc' program! PROGRAM EXITING!)
	 m2
o1000 endif
```

I get the following error at the o1000 line:
---
Expected CBRACKET_TOKEN found ID_TOKEN At: /var/lib/bbctrl/upload/notch_and_drill_drawer.nc:336:55
---
I counted the number of brackets over and over again, and my syntax and it seems correct.
![Screenshot_20191130-162749](https://user-images.githubusercontent.com/415031/69920938-87fb1e80-1442-11ea-9053-54747acebed9.png)

After playing around for awhile, I found that I could test two expressions with OR but at line position 55, it will always fail when adding additional expressions

**Comment by kwince** [comment]
Line 336 is the line beginning with o1000

I will include that in my big report later today

**Comment by jcoffland** [comment]
Note, one way to work around this problem is to add more brackets.  I.e.


```
o1000 if [  [ #5000 eq 0 ] OR [ [ #<_x_homed>  EQ 0] OR [ [ #<_y_homed> EQ 0]  OR [ #<_z_homed> EQ 0]  ] ] ]
   (msg, MACHINE NOT HOMED using 'zero.nc' program! PROGRAM EXITING!)
	 m2
o1000 endif
```

**Comment by jcoffland** [comment]
This has been fixed in [v0.4.12-rc21](https://buildbotics.com/bbctrl/bbctrl-0.4.12-rc21.tar.bz2).

**Comment by kwince** [comment]
Nice work thx 🙃
